pool/connection_pool_test.go

```diff
@@ -3495,6 +3503,39 @@ func runTestMain(m *testing.M) int {
 	return m.Run()
 }
 
+func TestConnectionPool_GetInfo(t *testing.T) {
+	t.Run("equal instance info", func(t *testing.T) {
+		var tCases [][]pool.Instance
+
+		tCases = append(tCases, makeInstances([]string{servers[0], servers[1]}, connOpts))
+		tCases = append(tCases, makeInstances([]string{
+			servers[0],
+			servers[1],
+			servers[3]},
+			connOpts))
+		tCases = append(tCases, makeInstances([]string{servers[0]}, connOpts))
+
+		for _, tc := range tCases {
+			ctx, cancel := test_helpers.GetPoolConnectContext()
+			connPool, err := pool.Connect(ctx, tc)
+			cancel()
+			require.Nilf(t, err, "failed to connect")
+			require.NotNilf(t, connPool, "conn is nil after Connect")
+
+			info := connPool.GetInfo()
+
+			var infoInstances []pool.Instance
+
+			for _, infoInst := range info {
+				infoInstances = append(infoInstances, infoInst.Instance)
+			}
+
+			require.ElementsMatch(t, tc, infoInstances)
+			connPool.Close()
```

Comment:
I think that `defer connPool.Clode()` is required here.

```suggestion
			connPool, err := pool.Connect(ctx, tc)
			cancel()
			require.Nilf(t, err, "failed to connect")
			require.NotNilf(t, connPool, "conn is nil after Connect")
			
			defer connPool.Close()

			info := connPool.GetInfo()

			var infoInstances []pool.Instance

			for _, infoInst := range info {
				infoInstances = append(infoInstances, infoInst.Instance)
			}

			require.ElementsMatch(t, tc, infoInstances)
```

Reply:
Closing the connection pool can be useful for gracefull closure; when the test completes, the allocated resources will inevitably be returned and the connection closed

Reply:
Certainly, sorry for the wrong tip.